src/components/partner/partner.service.ts

```diff
@@ -186,6 +194,28 @@ export class PartnerService {
     return await mapListResults(results, (dto) => this.secure(dto, session));
   }
 
+  async listProjects(
+    partner: Partner,
+    input: ProjectListInput,
+    session: Session
+  ): Promise<SecuredProjectList> {
+    const projectListOutput = await this.projectService.list(
+      { ...input, filter: { ...input.filter, partnerId: partner.id } },
+      session
+    );
+
+    const perms = await this.authorizationService.getPermissions({
+      resource: IProject,
+      sessionOrUserId: session,
+    });
+
+    return {
+      ...projectListOutput,
+      canRead: perms.member.canRead,
```

Comment:
```suggestion
      canRead: perms.projects.canRead,
```
But there's still some confusion between who owns the permission etc. since there are multiple ways to access this data/relation.